sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -144,6 +154,18 @@ public static ClientPipeline Create(ClientPipelineOptions? options = default)
 
         int perTryIndex = index;
 
+        if (isLoggingEnabled)
```

Comment:
Ah!  I see you are only adding the policy if logging is enabled.  We should figure out what precedence rules are for the following cases:
- I set pipelineOptions.LoggingOptions.IsLoggingEnabled false and don't provide a pipelineOptions.LoggingPolicy
- I set pipelineOptions.LoggingOptions.IsLoggingEnabled false and do provide a pipelineOptions.LoggingPolicy
- I set pipelineOptions.LoggingOptions.IsLoggingEnabled false and do provide a pipelineOptions.LoggingPolicy but pass it options.IsLoggingEnabled = true
- I set pipelineOptions.LoggingOptions.IsLoggingEnabled true and don't provide a pipelineOptions.LoggingPolicy
- I set pipelineOptions.LoggingOptions.IsLoggingEnabled true and do provide a pipelineOptions.LoggingPolicy
- I set pipelineOptions.LoggingOptions.IsLoggingEnabled true and do provide a pipelineOptions.LoggingPolicy but pass it options.IsLoggingEnabled = false

Is there a way to make this simple and obvious to the end-user what will happen?

Reply:
Hmmm yeah I hadn't thought this. I think the following are probably the most confusing cases:

- A library sets a custom logging policy on the options (maybe they want to set the event source name). A customer wants to disable it, so they set options.LoggingOptions.IsLoggingEnabled to false
- A customer sets options.LoggingOptions.IsLoggingEnabled to false thinking this will turn off System.ClientModel default logging, and provides their own logging policy instead

A good middle ground may be:
- If IsLoggingEnabled is false:
    - If options.LoggingPolicy is not null, still use it. BUT continue to check .IsLoggingEnabled in ClientLoggingPolicy, this would handle case 1 if the custom logging policy inherits from ClientLoggingPolicy
    - If options.LoggingPolicy is null, do not add any logging policy to the pipeline
- If IsLoggingEnabled is true:
    - Use options.LoggingPolicy if it's not null, otherwise set a default.

However, I think number 1 is probably going to be a bigger concern than 2 because all of the azure core clients will potentially be setting a custom logging policy, and this should be respected. So maybe clear documentation on IsLoggingEnabled saying "if you set this to false no logging policy will be added at all, even if you set one on the options" is better - I'll keep thinking about this.